services/supabase/gitauto_manager.py

```diff
@@ -69,6 +72,26 @@ def create_installation(
                 json={"owner_id": owner_id, "stripe_customer_id": customer_id}
             ).execute()
 
+        users_manager = UsersManager(client=self.client)
+        email_to_store = None if str(email).lower().endswith("@users.noreply.github.com") else email
+
+        response: APIResponse = self.client.table("users").select("email").eq("user_id", user_id).execute()
+
+        if response.data and len(response.data[0]) > 0:
+            # Update the email if it's different from the stored one
+            if email_to_store != response.data[0].get('email'):
+                self.client.table("users").update(
+                    json={"email": email_to_store}
+                ).eq("user_id", user_id).execute()
+        else:
+            # Create the user if it doesn't exist
+            users_manager.create_user(
+                user_id=user_id,
+                user_name=user_name,
+                installation_id=installation_id,
+                email=email_to_store,
+            )
```

Comment:
Why you remove 

```
self.client.table(table_name="users").upsert(
            json={
                "user_id": user_id,
                "user_name": user_name,
            },
            on_conflict="user_id",
        ).execute()
```

and handle data like above if else statement?

Reply:
because if no user with that user_id exists, it will create the user. if a user with that user_id does exist, it will update the email